### PR TITLE
add pt translation in i18n

### PR DESCRIPTION
**Pull request by ricardobr001**
## What does this PR?

This code add the `pt (portuguese)` translation in the i18n support :)

## Issue reference

Closes https://github.com/katherinepeterson/readme.so/issues/72

### CODE REVIEW

File: public/locales/pt/common.json
Comment (rohitjmathew) [inline public/locales/pt/common.json:9]: `Language` in portugese on google translate shows up as `língua`. Is this correct?

File: public/locales/pt/editor.json
Comment (rohitjmathew) [inline public/locales/pt/editor.json:11]: Do you think there is a more Portuguese word for Download, Editor, Preview and Raw? Asking incase it makes sense to translate those as well or leave them in native english itself

**Review by rohitjmathew — CHANGES_REQUESTED** [review]
The code looks good 💯 👍 🍻

Added a few comments. Do have a look

File: public/locales/pt/common.json
Comment (ricardobr001) [reply public/locales/pt/common.json:9]: It can be `Língua` too, but I think `Idioma` is more friendly, discord for example uses `Idioma` instead of `Língua` haha

File: public/locales/pt/editor.json
Comment (ricardobr001) [reply public/locales/pt/editor.json:11]: `Editor` have the same translation haha
`Download` we use download here for it, the translation for it is `baixar` but we rarely it here
`Preview` we can change it to `Pré visualização`
`Raw` we can change it to `código fonte`

**Review by rohitjmathew — APPROVED** [review]
Let's Get This Merged 👍 🚀  #LGTM

File: public/locales/pt/common.json
Comment (waldyrious) [inline public/locales/pt/common.json:4]: We don't have to use the anglicism here :)

```suggestion
  "description": "O nosso simples editor permite que você facilmente adicione e personalize todas as seções que você precisa para o README do seu projeto",
```

File: public/locales/pt/common.json
Comment (waldyrious) [inline public/locales/pt/common.json:6]: No need to capitalize (same as `get-started` in the line above)

```suggestion
  "100-free": "100% gratuito",
```

File: public/locales/pt/common.json
Comment (waldyrious) [inline public/locales/pt/common.json:14]: Same here, no need to capitalize:

```suggestion
  "translation-error": "Tradução incorreta?"
```

File: public/locales/pt/editor.json
Comment (waldyrious) [inline public/locales/pt/editor.json:2]:
```suggestion
	"editor-desktop-optimized": "Este site está otimizado para o desktop",
```

File: public/locales/pt/editor.json
Comment (waldyrious) [inline public/locales/pt/editor.json:5]:
```suggestion
	"download-readme-generated": "Readme gerado!",
```

File: public/locales/pt/editor.json
Comment (waldyrious) [inline public/locales/pt/editor.json:8]:
```suggestion
	"download-coffee": "Se você achou este projeto útil, considere apoiar-me com uma xícara de café!",
```

File: public/locales/pt/editor.json
Comment (waldyrious) [inline public/locales/pt/editor.json:10]: Typo :)

```suggestion
	"editor-select": "Selecione uma seção da barra lateral a esquerda para editar os conteúdos",
```

File: public/locales/pt/editor.json
Comment (waldyrious) [inline public/locales/pt/editor.json:11]:
```suggestion
	"preview-column-preview": "Pré-visualização",
```

File: public/locales/pt/editor.json
Comment (waldyrious) [inline public/locales/pt/editor.json:15]:
```suggestion
	"section-column-click-add": "Clique em uma seção abaixo para adicioná-la ao seu readme"
```

**Comment by ricardobr001** [comment]
Hi there @waldyrious, thanks for the review :)
There are some changes that are correctly, but some of them don't make any difference
I see that you're from Portugal, and there are some differences between the Portugal Portuguese to the Brazilian Portuguese, maybe I can change the translations for `portuguese (BR)`, WDYT?

**Comment by waldyrious** [comment]
I actually only suggested changes that I felt wouldn't make the text deviate from Brazilian Portuguese, but merely use a slightly less colloquial phrasing that would be less jarring for European Portuguese speakers. That way we wouldn't need to maintain two separate translations.

It wasn't clear to me whether you actively disagree with some of the proposed changes, or just feel that they are neutral and not worth making (i.e. not a clear improvement). Can you reply to the individual suggestions so we can try to find an agreement?

**Comment by ricardobr001** [comment]
> I actually only suggested changes that I felt wouldn't make the text deviate from Brazilian Portuguese, but merely use a slightly less colloquial phrasing that would be less jarring for European Portuguese speakers. That way we wouldn't need to maintain two separate translations.
> 
> It wasn't clear to me whether you actively disagree with some of the proposed changes, or just feel that they are neutral and not worth making (i.e. not a clear improvement). Can you reply to the individual suggestions so we can try to find an agreement?

Totally agreed with you :)
It'll be more easier to maintain only one translation haha
I'm going to accept the changes that I agree that need changes and reply for the other ones

File: public/locales/pt/common.json
Comment (ricardobr001) [reply public/locales/pt/common.json:4]: I don't think anglicism is a problem, and the `customize` sounds awkward in the European Portuguese?
I agree with the `README`

File: public/locales/pt/editor.json
Comment (ricardobr001) [reply public/locales/pt/editor.json:2]: I think this translation is ok, don't make much difference IMO

File: public/locales/pt/editor.json
Comment (ricardobr001) [reply public/locales/pt/editor.json:8]: I think this one is ok too
The `apoiar-me` sound too formal IMO

File: public/locales/pt/editor.json
Comment (ricardobr001) [reply public/locales/pt/editor.json:15]: This one don't make much difference IMO

**Comment by katherinepeterson** [comment]
> > I actually only suggested changes that I felt wouldn't make the text deviate from Brazilian Portuguese, but merely use a slightly less colloquial phrasing that would be less jarring for European Portuguese speakers. That way we wouldn't need to maintain two separate translations.
> > It wasn't clear to me whether you actively disagree with some of the proposed changes, or just feel that they are neutral and not worth making (i.e. not a clear improvement). Can you reply to the individual suggestions so we can try to find an agreement?
> 
> Totally agreed with you :)
> It'll be more easier to maintain only one translation haha
> I'm going to accept the changes that I agree that need changes and reply for the other ones

hey @ricardobr001, is this PR good to go? The code looks good, but I know there is a lot of discussion regarding the language

File: public/locales/pt/editor.json
Comment (waldyrious) [reply public/locales/pt/editor.json:15]: If it doesn't make it worse for pt-BR, please accept it then — for pt-PT, "adicionar no" is plain wrong.

File: public/locales/pt/editor.json
Comment (waldyrious) [reply public/locales/pt/editor.json:8]: I understand. However, "dar um suporte" sounds odd - would "dar suporte" work for you?

File: public/locales/pt/editor.json
Comment (waldyrious) [reply public/locales/pt/editor.json:2]: I know in pt-BR there "esse" and "este" can be used as synonyms, but for pt-PT it's the difference between "this" and "that" (the latter being wrong, of course). If it doesn't make a difference for pt-BR, please accept it then, since it does make a difference for pt-PT.

File: public/locales/pt/common.json
Comment (waldyrious) [reply public/locales/pt/common.json:4]: It's not awkward, just unnecessary. Out of curiosity, does "personalizar" sound odd to you, compared to "customizar"? Or is it just irrelevant?

**Comment by waldyrious** [comment]
@katherinepeterson most of the discussion has been resolved already, actually — we're just fleshing out the remaining expressions, so that the translation will be usable for both Brazilian Portuguese and European Portuguese. I'll let you know when things are all settled :)

File: public/locales/pt/common.json
Comment (ricardobr001) [reply public/locales/pt/common.json:4]: They're irrelevant, the two words have the same meaning haha
We can go with your suggestion :)

File: public/locales/pt/editor.json
Comment (ricardobr001) [reply public/locales/pt/editor.json:2]: Nice, didn't know that was a difference between `esse` and `este` in pt-PT, let's go with your suggestion

File: public/locales/pt/editor.json
Comment (ricardobr001) [reply public/locales/pt/editor.json:8]: I would go with `me apoiar`, WDYT?

File: public/locales/pt/editor.json
Comment (ricardobr001) [reply public/locales/pt/editor.json:15]: Nice, going to accept it :)

**Comment by ricardobr001** [comment]
When I updated my code here and pushed to my fork, something made this PR look strange 🤔
There are strange additions and strange conflicts
I'm going to close it and open another one, ok? @waldyrious @katherinepeterson

**Comment by waldyrious** [comment]
@ricardobr001 I'd be happy to help you restore the PR into a working state. If you add me as a collaborator to your fork I should be able to help. Let me know if that's OK!

**Comment by ricardobr001** [comment]
> 
> 
> @ricardobr001 I'd be happy to help you restore this PR into a working state. If you add me as a collaborator to your fork I should be able to help. Let me know if that's OK!

Nice, going to add you there @waldyrious 
I tried here to restore the code, but when I go to open a new PR, the github changes is saying that I'm adding the italian translations, but they're already there, this is something strange 🤔

**Comment by ricardobr001** [comment]
@waldyrious I managed here to fix this, the problem was that when I was updating my fork, I was using the rebase and merge, I've changed it to merge commit and the problem of re-adding files is gone :)

File: public/locales/pt/editor.json
Comment (waldyrious) [reply public/locales/pt/editor.json:8]: Sounds good to me!

**Comment by ricardobr001** [comment]
@waldyrious Updated the translation with what we have discussed
@katherinepeterson Updated the code with the changes made in the main branch

**Review by waldyrious — APPROVED** [review]
Great work @ricardobr001! LGTM 👍

**Comment by MicahBCode** [comment]
To improve the PR you could link the issue using a keyword following the instructions on here: https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue

This will result in automatically closing the issue after merging.
Maybe you can edit your first comment adding the keyword?

**Comment by ricardobr001** [comment]
Thanks for the tip @MicahBCode didn't know about that haha
Also I have update the code here :)

File: public/locales/pt/common.json
Comment (waldyrious) [inline public/locales/pt/common.json:14]: Heads-up: these entries were [removed in #96](https://github.com/katherinepeterson/readme.so/pull/96/files#diff-4840952983d2a94f27deb00bf7247347345b60e3a96dc2aa41a6a50ecacdf1b6).

Comment (ricardobr001) [reply public/locales/pt/common.json:14]: Ops, going to remove it haha

**Comment by ricardobr001** [comment]
@katherinepeterson I think we can merge this, WDYT?

**Comment by ayrtonvwf** [comment]
Hey folks. I was hoping to contribute with the portuguese translation but found out it's already done in this PR 😁
I'd just like to point out that if this is Brazilian Portuguese we should tag it as pt-BR as per [BCP47](https://www.techonthenet.com/js/language_tags.php) thus allowing for other variants and using a standard notation that would make the internationalization files/code future proof and providing clear guidelines for the next contributions ❤️

**Comment by waldyrious** [comment]
@ayrtonvwf you might want to check out the discussion starting [here](https://github.com/katherinepeterson/readme.so/pull/73#issuecomment-825669469), where OP and I agreed that we could use a relatively neutral form of Portuguese that would allow us to avoid having to keep two translations that would be pretty much identical anyway. Let me know what you think about that!

**Comment by ricardobr001** [comment]
I think it would be better to maintain only one translation, with this, if in the future the application have a new phrase, we only add it in one file, the same logic applies if we remove a phrase, WDYT @ayrtonvwf ?

**Comment by ayrtonvwf** [comment]
I looked at the changes on this PR and realized it just translates the UI of the tool, not the generated README, right?

I'm sure it's not a problem to have only one translation for all variants in the UI. It won't feel as natural but users would benefit way more from faster updates (because of less burden to maintain the project), so that also seems right to me.

But what about the generated README? Will that be translated or english is set to be the only target language for the generated content? Asking because I'm not aware of previous thoughts on that.

**Comment by ricardobr001** [comment]
> I looked at the changes on this PR and realized it just translates the UI of the tool, not the generated README, right?
> 
> I'm sure it's not a problem to have only one translation for all variants in the UI. It won't feel as natural but users would benefit way more from faster updates (because of less burden to maintain the project), so that also seems right to me.
> 
> But what about the generated README? Will that be translated or english is set to be the only target language for the generated content? Asking because I'm not aware of previous thoughts on that.

Hmm, when I've opened this PR the templates aren't translated, but I see now there is a translation of the `.md` files to Turkish, this is something we can add, we can create a new issue for that or add in this PR

**Pull request merged**
